yjit/src/codegen.rs

```diff
@@ -6227,6 +6126,155 @@ fn gen_send_iseq(
     Some(EndBlock)
 }
 
+fn exit_if(asm: &mut Assembler, pred: bool, counter: Counter) -> Option<()> {
```

Comment:
Should comment to explain what this is used for. Put yourself in the shoes of someone who's never seen this codebase and maybe is relatively new to Rust.